napalm_brocade/brocade.py

```diff
@@ -0,0 +1,428 @@
+#
+# Copyright 2016 Shiv Haris, Brocade Communication Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License.  You may obtain a copy of
+# the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+"""
+Brocade-Napalm Driver.
+
+This driver is meant for SLX and NOS based switches.
+"""
+from netmiko import ConnectHandler
+from napalm_base.base import NetworkDriver
+from napalm_base.exceptions import ConnectionException, MergeConfigException, \
+    ReplaceConfigException, SessionLockedException, CommandErrorException
+
+import os
+import re
+from shutil import copyfile
+
+
+# TBD(shh) Put this in config file (oslo_config)
+EXPORT_HOST = "10.24.88.6"
+EXPORT_USER = "shh"
+EXPORT_PASSWORD = "ss"
+
+class BrocadeDriver(NetworkDriver):
+    """Napalm Driver for Vendor Brocade."""
+
+    def __init__(self, hostname, username, password, timeout=60,
+                 optional_args=None):
+        """
+        CTOR for the device.
+        """
+
+        if optional_args is None:
+            optional_args = {}
+
+        self.device = None
+        self.hostname = hostname
+        self.username = username
+        self.password = password
+        self.timeout = timeout
+        self.port = optional_args.get('port', 22)
+
+
+    def open(self):
+        """Open a connection to the device."""
+        try:
+            self.device = ConnectHandler(device_type='brocade_vdx',
+                                         ip=self.hostname,
+                                         port=self.port,
+                                         username=self.username,
+                                         password=self.password,
+                                         timeout=self.timeout,
+                                         verbose=True)
+        except Exception:
```

Comment:
Catch something less generic than ```Exception``` - ```NetMikoAuthenticationException``` and ```NetMikoTimeoutException``` is probably what you are looking for.